packages/keyring/src/pair/index.ts

```diff
@@ -124,8 +128,9 @@ export default function createPair ({ toSS58, type }: Setup, { publicKey, secret
       }
     },
     derive: (suri: string, meta?: KeyringPair$Meta): KeyringPair => {
-      assert(secretKey, 'Cannot derive on a locked keypair');
-
+      if (!secretKey || isLocked(secretKey)) {
+        throw new Error('Cannot derive on a locked keypair');
+      }
```

Comment:
```suggestion
      assert(!isLocked(secretKey), 'Cannot derive on a locked keypair');
      
```

isLocked already checks for !secretKey

Reply:
Done